Lib/test/test_unittest/test_program.py

```diff
@@ -191,6 +199,18 @@ def test_ExitEmptySuite(self):
         out = stream.getvalue()
         self.assertIn('\nNO TESTS RAN\n', out)
 
+    def test_ExitSetUpClassFailureSuite(self):
+        stream = BufferedWriter()
+        with self.assertRaises(SystemExit) as cm:
+            unittest.main(
+                argv=["setup_class_failure"],
+                testRunner=unittest.TextTestRunner(stream=stream),
+                testLoader=self.TestLoader(self.SetUpClassFailure))
+        self.assertEqual(cm.exception.code, 1)
+        out = stream.getvalue()
+        self.assertIn("ERROR", out)
```

Comment:
Let's check `"ERROR: setUpClass"` and `"SetUpClassFailure"` here. Otherwise I think it's good!